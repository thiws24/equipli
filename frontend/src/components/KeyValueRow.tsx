import * as React from "react"

interface Props {
  label: string | React.ReactNode
  children: React.ReactNode
}

export const KeyValueRow: React.FC<Props> = ({ label, children }) => {
  return (
    <div className="px-4 py-6 sm:grid sm:grid-cols-4 sm:gap-4 sm:px-0">
      <dt className="text-sm leading-6 text-customBlack sm:col-span-1">
        {label}
      </dt>
      <dd className="mt-1 text-sm font-medium leading-6 text-customBlack sm:col-span-3 sm:mt-0 sm:mr-8">
        {children}
      </dd>
    </div>
  )
}
