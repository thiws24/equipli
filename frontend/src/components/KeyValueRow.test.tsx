import { render, screen } from "@testing-library/react"
import { KeyValueRow } from "./KeyValueRow"
import { test, describe, expect } from 'vitest'

describe("KeyValueRow Tests", () => {
    test("correct rendering", () => {
        const label = "Test Label"
        const children = "Test Children"

        render(<KeyValueRow label={label}>{children}</KeyValueRow>)

        expect(screen.getByText(label)).toBeInTheDocument()
        expect(screen.getByText(children)).toBeInTheDocument()
    })

    test("renders label as a ReactNode", () => {
        const label = <span>ReactNode Label</span>
        const children = "Test Children"

        render(<KeyValueRow label={label}>{children}</KeyValueRow>)

        expect(screen.getByText("ReactNode Label")).toBeInTheDocument()
    })

    test("correct styles applied", () => {
        const label = "Styled Label"
        const children = "Styled Children"

        render(<KeyValueRow label={label}>{children}</KeyValueRow>)

        const labelElement = screen.getByText(label)
        expect(labelElement).toHaveClass(
            "text-sm leading-6 text-customBlack sm:col-span-1"
        )

        const childrenElement = screen.getByText(children)
        expect(childrenElement).toHaveClass(
            "mt-1 text-sm leading-6 text-customBlack sm:col-span-3 sm:mt-0 sm:mr-8"
        )
    })

    test("renders with empty children", () => {
        const label = "Empty Children"
        const children = null
        const { container } = render(
            <KeyValueRow label={label}>{children}</KeyValueRow>
        )

        expect(screen.getByText("Empty Children")).toBeInTheDocument()

        // Überprüfen, ob das `dd`-Element existiert, aber keinen Inhalt hat
        const ddElement = container.querySelector("dd")
        expect(ddElement).toBeInTheDocument()
        expect(ddElement).toBeEmptyDOMElement()
    })

    test("matches the snapshot", () => {
        const { container } = render(
            <KeyValueRow label="Snapshot Label">Snapshot Content</KeyValueRow>
        )
        expect(container.firstChild).toMatchSnapshot()
    })
})
