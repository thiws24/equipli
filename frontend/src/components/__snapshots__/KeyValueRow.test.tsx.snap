// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`KeyValueRow Tests > matches the snapshot 1`] = `
<div
  class="px-4 py-6 sm:grid sm:grid-cols-4 sm:gap-4 sm:px-0"
>
  <dt
    class="text-sm leading-6 text-customBlack sm:col-span-1"
  >
    Snapshot Label
  </dt>
  <dd
    class="mt-1 text-sm font-medium leading-6 text-customBlack sm:col-span-3 sm:mt-0 sm:mr-8"
  >
    Snapshot Content
  </dd>
</div>
`;

exports[`KeyValueRow Tests matches the snapshot 1`] = `
<div
  class="px-4 py-6 sm:grid sm:grid-cols-4 sm:gap-4 sm:px-0"
>
  <dt
    class="text-sm leading-6 text-customBlack sm:col-span-1"
  >
    Snapshot Label
  </dt>
  <dd
    class="mt-1 text-sm font-medium leading-6 text-customBlack sm:col-span-3 sm:mt-0 sm:mr-8"
  >
    Snapshot Content
  </dd>
</div>
`;
